zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/ElasticsearchArchiverRepository.java

```diff
@@ -185,6 +188,33 @@ public CompletableFuture<Void> reindexDocuments(
         .thenApplyAsync(ignored -> null, executor);
   }
 
+  private CompletableFuture<List<String>> fetchMatchingIndexes(final String indexWildcard) {
+    final Pattern indexNamePattern = Pattern.compile(indexWildcard);
+    return client
+        .indices()
+        .get(new Builder().index(ALL_INDICES).build())
+        .thenApply(
```

Comment:
:x: Use `thenApplyAsync` and specify the executor, that way we will not execute the callback if the background task manager is shut down.